### ISSUE TITLE
Next feature release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#7534]
Just a few thoughts about a possible next release:

We might have to release soon, depending on what happens to the live map next week (see related issue).
The fix for this however is currently only on master.
On master however I do see a regression of the mapsforge map (bumpy zooming, see related issue).

So in case we are forced to release I would prefer to merge master to release but downgrade mapsforge on release to the older version (if this is the cause for the bumpy zooming).

Other thoughts?

**Comment by moving-bits** [comment]
If we have to release pretty soon we can also cherry-pick the live map changes only to release branch.

**Comment by jonas-koeritz** [comment]
I've been experiencing several issues with the mapsforge map (on master), the zooming really doesn't work that great and my map style doesn't work at all, this should be an issue with the styles version and the mapsforge update.

**Comment by Lineflyer** [comment]
> I've been experiencing several issues with the mapsforge map (on master), the zooming really doesn't work that great 

Yes, see #7523 

>  and my map style doesn't work at all, this should be an issue with the styles version and the mapsforge update.

I am using openandromaps with elevate4 theme, which works same as with the older mapsforge version. What are you using and what means "does not work". If you have more details, feel free to create a dedicated issue for that. :)

**Comment by jonas-koeritz** [comment]
Elevate 4 works like a charm. I have previously been on Elevate 2 XL.

**Comment by Lineflyer** [comment]
> If we have to release pretty soon we can also cherry-pick the live map changes only to release branch.

I don't think its necessary as the nightlies are working good except the issues mentioned above.
How about merging master to release and just reverting the mapsforge version on release?

**Comment by Lineflyer** [comment]
> I think we need to make sure that a new user knows that he needs to supply credentials/have an account.

Already today the live map will more or less not deliver useful information without a login. Its even better, that now it does not deliver anything, so (at least the already experienced) users will easier see the problem, that he is not logged in.

Independently we need a better welcoming for first-time users, which is currently discussed in another issue. So I would suggest not to make this a hard prerequisite for releasing the new map.

**Comment by moving-bits** [comment]
> > If we have to release pretty soon we can also cherry-pick the live map changes only to release branch.
> 
> I don't think its necessary as the nightlies are working good except the issues mentioned above.
> How about merging master to release and just reverting the mapsforge version on release?

In this case (stable nightlies) this will work as well. (And is even better, because more feature-rich.)
The other way was just a suggestion for a hotfix in case the nightlies were either unstable or untested.

**Comment by Lineflyer** [comment]
@rsudev @UniQP 
If one of you has time today, I would be fine with merging the current state of `master` onto `release` today.
Afterwards I can take care of reducing the mapsforge version and handling the changelog on `release`.

**Comment by Lineflyer** [comment]
@rsudev 
I thought the mapsforge version change would be a one-liner in build.gradle. Just now I see, that it is a bit more which would need to be reverted. Probably better if you could have a look ;)

**Comment by jonas-koeritz** [comment]
I would like to see https://github.com/cgeo/cgeo/pull/7545 included with the release to avoid the strange behavior we documented in #7539.

**Comment by Lineflyer** [comment]
> I would like to see #7545 included with the release to avoid the strange behavior we documented in #7539.

Yes. Either we merge this PR before branching or we can later target it onto release. Lets see.

**Comment by Lineflyer** [comment]
The announced map change seems to be live.
From what I can see, the old map is still under same URL and both types are coexisting as before. In fact only some navigation targets on the website have been changed to priorize the new map.

However it is for sure only a matter of time until the old map gets retired, but we have seemingly a bit time to move over.

**Comment by Lineflyer** [comment]
Website tests are also green and manual testing shows normal behaviour.

Still I would like to give out current master as a beta version, which means merging at some point of time in near future would be appreciated. However we can probably merge some ongoing fixes e.g. for the new map.

**Comment by Lineflyer** [comment]
I will be on vacation 01.05. to 14.05. 
If we want to bring out a beta version from master the merge should be done before (or someone else needs to take care of it).

**Comment by rsudev** [comment]
I have the feeling that current master is a bit 'unstable' due to small glitches (that includes the performance issue in some context in mf 0.11).
I would follow your suggestion to revert this change for the moment.
Having that chain of chnages on a beta for your vacation (if that is the plan) sounds like a good idea to get better feedback for stabilizing the next release.
Perhaps this timeframe will give me enough time to get the long awaited new map fixes also in shape.

**Comment by Lineflyer** [comment]
> I have the feeling that current master is a bit 'unstable' due to small glitches

Anything special you have in mind ?

> Having that chain of chnages on a beta for your vacation (if that is the plan) sounds like a good idea to get better feedback for stabilizing the next release.

So we should maybe have a last look over the submitted but not merged PRs and then branch to `release`?

> Perhaps this timeframe will give me enough time to get the long awaited new map fixes also in shape.

From support channels a major point is #7209

**Comment by Lineflyer** [comment]
Regarding the open PR:

IMHO all I would like to see in next release are merged.
For #7439, #7441 and #7579 I would like to have more time for testing on master (so not in next release).

@rsudev 
If you want me to publish a beta before my vacation we should merge branches today, because only two days left until vacation.

**Comment by Lineflyer** [comment]
In parallel we should upload strings to Crowdin as we have a bunch of new or modified ones.

**Comment by rsudev** [comment]
#7579 we can merge after the beta to release without big risk (I made an alternative suggestion to fix the layout glitches in a 'safer' fashion).
I can upload to crowdin and merge to release this evening probably.

**Comment by moving-bits** [comment]
I've adjusted #7579 according to rsudev's proposal.

**Comment by rsudev** [comment]
I would be in favour of including it now in the beta - @Lineflyer, what do you think?

**Comment by Lineflyer** [comment]
Sure

**Comment by Lineflyer** [comment]
Done

**Comment by Lineflyer** [comment]
For your attention:

@rsudev merged `master` into `release`.
Fixes for regressions of a possible next release/for problems found in the beta version should therefore now target `release` if applicable.

Strings have also been uploaded to Crowdin and need to be downloaded before releasing (not necessarily for the beta version).

I will smoke test the RC and publish it soon if smoke test is ok.

**Comment by Lineflyer** [comment]
2019.04.29-RC smoke tested (map, save, view, search, log) on devices with Android 5.0.1, 4.1.2 and 6.0.1 and just now published onto the beta channel.

If release is needed during my vacation, one should look onto the android vitals and beta feedback on Google Play Console to check for problems.
Checklist for releases can be found [here](https://github.com/cgeo/cgeo/wiki/Checklist-for-release-process) and should contain all relevant steps for a good release process.

**Comment by Lineflyer** [comment]
There are more users now on support mail with the problem, that TB search fails in the current beta version.
As I cannot test the related fix (as I cannot reproduce the problem) I plan to release another beta version.

Anything else which should get into the next beta?

**Comment by Lineflyer** [comment]
Addition:
Will do this next week after return from holiday.

**Comment by bekuno** [comment]
Please add https://github.com/cgeo/cgeo/commit/6d3ece2dff04ebeaf77cc899da22bd2ffb548e6e
to release.

**Comment by Lineflyer** [comment]
This change is currently only on `master`. I also do not see it as critical for the next release, as it is no bug fix.

I did just 30 minutes ago already smoke test and publish a new beta version based on current `release` branch status. I will contact some beta users to check whether the travelbug problem is gone.

**Comment by Lineflyer** [comment]
FYI:
I consider the current release blocked by #7617 which is a regression in 2019.05.13-RC

**Comment by Lineflyer** [comment]
As a workaround was applied for #7617 I think we can release. Will do so tonight unless someone does see a problem for releasing.

**Comment by rsudev** [comment]
Shall I sync translations from crowdin before?

**Comment by Lineflyer** [comment]
@rsudev That would be great!

**Comment by Lineflyer** [comment]
Preparing release now:

## Release checklist for full release:

- [x] Check/Import incoming Translations from Crowdin
- [ ] Check/Insert correct version name as headline in ```changelog-release.xml```
- [ ] Build release on CI server
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by Lineflyer** [comment]
@kumy @mucek4 
There seems to be a problem signing the APK on CI for the Fdroid Release Publish Job:
http://ci.cgeo.org/view/FDroid/job/Publish%20FDroid%20-%20RELEASE/45/console

Can one of you take a look?
Also I do not see new nightlies in the packet repo since 2019.05.18 (2 days ago)

**Comment by kumy** [comment]
I'll give a quick look on Fdroid job

**Comment by kumy** [comment]
It should be fixed thanks to https://github.com/cgeo/cgeo-fdroid-base/commit/001460415c3f34866752541208701ec2480f4c10. I've pushed a second commit that should mark job as failed in on any error.

**Comment by Lineflyer** [comment]
Thanks @kumy 
Looks better now!

@cgeo/team 
Update has been uploaded to Play, but play still digests it. Its not yet available to download.
See checklist above for current status. Will check back tomorrow.

**Comment by Lineflyer** [comment]
Just had the chance to check back and release is available on Play, so we are done!
Thanks to all people involved and special thanks to @jonas-koeritz for the live map implementation!!

I will try to monitor the performance in the field next days.

**Comment by bekuno** [comment]
I miss the the cool hint button feature in the release notes.

**Comment by Lomanic** [comment]
cgeo update is still not showing up in f-droid for me. Tried disabling and re-enabling the cgeo repository and reloading (with the pull down gesture) without success.

**Comment by kumy** [comment]
@Lomanic :( will look in the evening

**Comment by fgerling** [comment]
The update showed up in fdroid. I'm not sure _why_.   
I disabled and enabled the repository again some minutes ago.

**Comment by kumy** [comment]
@fgerling @Lomanic As said by @fgerling Disabling / enabling the source fixed the problem for me. Seems to be an fdroid issue, no?

**Comment by Lineflyer** [comment]
@cgeo/team 
First check on Google Play crash reports shows no suspect things so far. Also support mail is relatively quiet besides some people now realizing, that they have never been logged in (as now there are no more caches shown on live map) which is positive.

Lets wait for the weekend, but looks rather good! Thanks again for your work!
